uPortal-web/src/main/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDao.java

```diff
@@ -302,4 +305,53 @@ public void aggregateMarketplaceRating() {
             }
         }
     }
+    
+    @Override
+    public Set<IMarketplaceRating> getAllRatingsForPortletInLastXDays(int numberOfDaysBack, final String fname) {
+        
+        final Calendar queryDate = Calendar.getInstance();
+        queryDate.add(Calendar.DAY_OF_YEAR, 0-numberOfDaysBack);
```

Comment:
:question: Could the Java 8 Time processing utils be used to simplify/clarify the code?
http://www.baeldung.com/migrating-to-java-8-date-time-api

Reply:
You're correct could use LocalDateTime minusDays method:
https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html#minusDays-long- 

not sure what to do to modify this given the submited pull request

Reply:
> to modify this given the submited pull request

When more changes are pushed to a git branch, those changes are automatically included in a pull request.